pkg/util/azureclient/azuresdk/armnetwork/subnets.go

```diff
@@ -7,24 +7,23 @@ import (
 	"context"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
-	sdknetwork "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
 
 	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
 )
 
 // SubnetsClient is a minimal interface for azure-sdk-for-go subnets client
 type SubnetsClient interface {
-	Get(ctx context.Context, resourceGroupName, virtualNetworkName, subnetName string, options *sdknetwork.SubnetsClientGetOptions) (sdknetwork.SubnetsClientGetResponse, error)
+	Get(ctx context.Context, resourceGroupName, virtualNetworkName, subnetName string, options *armnetwork.SubnetsClientGetOptions) (armnetwork.SubnetsClientGetResponse, error)
+	SubnetsClientAddons
 }
 
 type subnetsClient struct {
-	*sdknetwork.SubnetsClient
+	*armnetwork.SubnetsClient
 }
 
-var _ SubnetsClient = (*subnetsClient)(nil)
```

Comment:
Not a concern I'd block merging over, but why'd you remove this line and the corresponding line for the vnet client?

Reply:
This idiom is to check if the struct implements the interface in the compile time.
This is usually used in libraries where there are some exposed structs which are not used within the same libraries.

In our case, the struct is assigned to some variables (or struct fields) whose type is the interface, so the implementation is validated at that point, so we don't have to check here.

https://go.dev/doc/effective_go#blank_implements
> By convention, such declarations are only used when there are no static conversions already present in the code, which is a rare event.